### PR TITLE
[@labs/react] update react to 18

### PR DESCRIPTION
**Pull request by taylor-vann**
Update react dependencies to 18

### CODE REVIEW

File: packages/labs/react/src/test/create-component_test.tsx [hunk 1]
```diff
@@ -4,14 +4,17 @@
  * SPDX-License-Identifier: BSD-3-Clause
  */
 
-import type * as ReactModule from 'react';
+// import type * as ReactModule from 'react';
```
Comment (sorvell) [inline packages/labs/react/src/test/create-component_test.tsx:7]: Are these changes intended? If not, revert.

**Review by sorvell — CHANGES_REQUESTED** [review]
In addition to comments, please add a changeset and ensure all checks pass. Thanks.

**Comment by augustjk** [comment]
Superseded by https://github.com/lit/lit/pull/4000
